### PR TITLE
fix: azure blob file not found

### LINKED ISSUE #11673: Plus signs in Media file or folder name break Media Library links and previews
### Describe the bug
if i upload a file named `aaa+++.jpg` to azure storage. it will lost in preview page.

### To Reproduce
1. Upload a file named `aaa+++.png` in media library page.
2. File name become a `aaa   .png` in preview page. but the real filename is `aaa+++.png`
3. So the file not found.

### Expected behavior
the filename is `aaa+++.png` in preview page

### Screenshots
image
image

### PR DESCRIPTION
**Pull request by IsQiao**
Fixes #11673

### CODE REVIEW

**Comment by Skrypt** [comment]
The fix is questionnable. Needs investigation.

File: src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs [hunk 1]
```diff
@@ -145,7 +145,7 @@ await foreach (var blob in page)
                 }
                 else
                 {
-                    var itemName = Path.GetFileName(WebUtility.UrlDecode(blob.Blob.Name)).Trim('/');
+                    var itemName = Path.GetFileName(blob.Blob.Name).Trim('/');
```
Comment (deanmarcussen) [inline src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:148]: The decode code is old from the previous blob library from Microsoft.

It maybe that it is no longer delivered by the azure library encoded, so a decode doesn't do anything useful.

Comment (IsQiao) [reply src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:148]: the `decode` can damage file path. `aaa+++.jpg` -> `aaa   .jpg`
remove `decode` can resolve it if use Kestrel Server directly.
but need to set `allowDoubleEscaping="true"` in web.config if use IIS Server.

File: src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs
Comment (Piedone) [inline src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:1]: Please check the decode in `GetDirectoryContentFlatAsync()` too.

While this indeed fixes the linked bug, I'm not sure if it'll break anything else. Can you upload a file to Blob Storage with a name that has all kinds of characters that should be encoded for a URL, and it'll keep them in their original form?

**Review by Piedone — CHANGES_REQUESTED** [review]
I sincerely apologize for us taking so much time here. I checked out the PR, going over old ones, and this would be quite useful.

Is this something you'd like to revisit any time soon or should we close?

File: src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs
Comment (IsQiao) [reply src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:1]: ok, i will to test all kinds of chars for this case and resolve the PR conflicts. please wait.

Comment (Piedone) [reply src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:1]: Thank you!

**Comment by Piedone** [comment]
Not to be a bother, but any news here, @IsQiao?

**Comment by IsQiao** [comment]
Sorry, I found new bugs regarding this fix and I will fix them.

**Comment by Piedone** [comment]
Great, thanks! When you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

**Comment by IsQiao** [comment]
> Great, thanks! When you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

ok, i will click the button after fix the bug. thanks

**Comment by Piedone** [comment]
This needs to be fixed not just for Blob Storage, see https://github.com/OrchardCMS/OrchardCore/issues/11673#issuecomment-2134038340.

**Pull request closed without merging**